parcels/field.py

```diff
@@ -864,19 +880,20 @@ def advancetime(self, field_new, advanceForward):
     def computeTimeChunk(self, data, tindex):
         g = self.grid
         with NetcdfFileBuffer(self.dataFiles[g.ti+tindex], self.dimensions, self.indices, self.netcdf_engine) as filebuffer:
```

Comment:
what is `self.dataFiles` in the case of an xarray dataset? What happens if there are no data files, like in your test below?

Reply:
These `self.dataFiles`refer to the Dataset (`ds`) itself. `Filebuffer.dataFiles` is set in field.py 
https://github.com/OceanParcels/parcels/blob/1020717d363426c49a2425d445b298d65f910adb/parcels/field.py#L208-L211

And `dataset` is set at the top of the file as 
https://github.com/OceanParcels/parcels/blob/1020717d363426c49a2425d445b298d65f910adb/parcels/field.py#L158-L162

This may not be the cleanest way to do it (and the variable names may be confusing, as they originally referred to actual NetCDF files), but this was a simple way to implement xarray handling too.